### PR TITLE
fix: evaluate integer percentage

### PR DESCRIPTION
**Pull request by rayzhou-bit**
Currently, an integer percentage answer like "1%" does not evaluate as a number while "1.0%" does. This is due to the % suffix not being evaluated when there is no "." in the answer. This PR fixes this issue.

### CODE REVIEW

**Review by jesperhodge — CHANGES_REQUESTED** [review]
Nice work identifying the bug! The library does not seem straightforward to me, it seems a bit overengineered to go through all this trouble just to parse a percentage number ^^ so it's great you figured this out!

I'm not sure how you tested this, but it's a library, right? Typically we would follow Test-Driven Development practices to make sure that the library behaves correctly, since we don't test libraries manually so much.

Generally I think doing red-green testing on this would be a good approach: first writing a failing test that properly catches the bug you found, and only afterwards writing the code that fixes it until the test becomes green.

You'd also want to think about every conceivable edge case, but I can't come up with any relevant edge case here, so I think it's just a matter of writing that new test and we should be good.

Another thing to consider is why this bug occurred. I mean this is a very standard operation, right? Converting a string that may include percentage sign into a float. In my opinion this should not have occurred since in a mathematical library there are very defined operations.

Can we answer these questions:
- is this library sufficiently tested? I mean are we confident the tests will catch whatever we throw at it and this was just one unfortunate exception? Or are the tests just not sufficient and we should reevaluate them?
- did anything about the way the code is written make this bug likely to happen? Why is it there is a bug in such an apparently straightforward operation? What could we improve to avoid bugs like this going forward?

File: calc/calc.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def lower_dict(input_dict):
 
 def super_float(text):
     """
-    Like float, but with SI extensions. 1k goes to 1000.
+    Evaluate suffixes if applicable and apply float.
```
Comment (jesperhodge) [inline calc/calc.py:115]: I would like to still see `SI extensions` mentioned and some example would also be helpful in the comment.

File: calc/calc.py [hunk 2]
```diff
@@ -128,7 +128,7 @@ def eval_number(parse_result):
     Calls super_float above.
     """
     for item in parse_result:
-        if "." in item or "e" in item or "E" in item:
+        if "." in item or "e" in item or "E" in item or parse_result[-1] in SUFFIXES:
```
[the remark this answers is not in the document]
Comment (jesperhodge) [reply calc/calc.py:131]: why not `"%" in item`? Not a suggestion, I genuinely try to understand
I also don't think suffixes necessarily only contains percentage sign, so the check here doesn't seem very logical.
And... is "e" and "%" the only SI extensions that could be relevant? What if someone adds another extension?
Definitely some code smells for me.

**Comment by jesperhodge** [comment]
I have to say I find it quite suspicious that we need to write our own, obviously buggy, library to convert something like a percentage string into a float. Isn't there a more properly tested and more reliable python package that does that that we know is good?

**Comment by rayzhou-bit** [comment]
@jesperhodge I did test the code through tests on `edx-platform`. I'm realizing now I can incorporate tests in this repo as well so I will work on adding that. 

The way the code is written here, I'm expecting many more `SUFFIXES` that can be applied outside of just `%`. One such suffix would be `k`, which the comments mention but isn't actually used.

I am very confident in this fix because the specific issue is that `1.0%` works as an answer while `1%` does not. When `eval_number` receives an input of `['1', '%']`, it errors on `int("".join(parse_result))` because the `%` suffix wasn't evaluated in super_float. Meanwhile, an input of `['1', '.', '0', '%']` does evaluate `%` because there is a `.`.

File: calc/calc.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rayzhou-bit) [reply calc/calc.py:131]: Looking at this again, it looks like previously the code was to simply always apply `super_float` and was changed to have this conditional in a python 3.11 support PR. The idea behind the condition is to apply `super_float` if the number has a decimal, has scientific notation or a suffix (missing before this PR). I agree it's not ideal and I'll think about a more elegant solution.

**Comment by rayzhou-bit** [comment]
Interesting note that unit test for -70% = -0.7 does not work while other numbers do.

> AssertionError: -0.7000000000000001 != -0.7 within -7e-07 delta (1.1102230246251565e-16 difference) : Failure in testing suffix '%': '-70%' was not -0.7

File: calc/calc.py [hunk 3]
```diff
@@ -79,7 +79,7 @@
 }
 
 SUFFIXES = {
```
Comment (jesperhodge) [inline calc/calc.py:81]: `SUFFIXES` is used in one other place. I'm surprised it didn't break any tests.

Let's make sure we don't break existing logic.

You could leave this dict as it was originally and add a second dict, for example:
```
SUFFIX_EVALS = {
    '%': lambda x: x * SUFFIXES['%']
}
```

**Review by jesperhodge — CHANGES_REQUESTED** [review]

File: calc/calc.py (hunk 3 above)
Comment (rayzhou-bit) [reply calc/calc.py:81]: I looked at the other uses of `SUFFIXES` and it should be fine. The other uses are only checking the key of the object to find what should be considered a suffix and not using the value in the object to evaluate anything.

**Review by jesperhodge — APPROVED** [review]
